virttest/iscsi.py

```diff
@@ -833,5 +835,494 @@ def create_iSCSI(params, root_dir=data_dir.get_tmp_dir()):
+
+        :return: True if existed.
+        :rtype: Boolean
+        """
+        cmd = "targetcli ls /backstores/%s" % self.iscsi_backend
+        output = process.run(cmd).stdout_text
+        if " %s " % self.backstore_name in output:
+            return True
+        return False
+
+    def create_backstore(self):
+        """
+        Create fileio in backstore.
+        """
+        if not self._existed():
+            # create image disk
+            if not os.path.isfile(self.emulated_image):
+                process.system(self.create_cmd)
+            else:
+                emulated_image_size = os.path.getsize(self.emulated_image) // 1024
+                if emulated_image_size != self.emulated_expect_size:
+                    # No need to remove, rebuild is fine
+                    process.system(self.create_cmd)
+
+            # Create a backstore
+            backstore_cmd = "targetcli /backstores/%s/ create %s %s" % (
+                self.iscsi_backend,
+                self.backstore_name,
+                self.emulated_image,
+            )
+            process.run(backstore_cmd)
+
+            # Set attribute
+            if self.iscsi_lun_attrs:
+                attr_cmd = "targetcli /backstores/%s/%s set attribute %s" % (
+                    self.iscsi_backend,
+                    self.backstore_name,
+                    self.iscsi_lun_attrs,
+                )
+                process.system(attr_cmd)
+
+            # Save configuration
+            process.system("targetcli / saveconfig")
+
+    def delete_backstore(self):
+        """
+        Delete fileio in backstore.
+        """
+        if self._existed():
+            backstore_cmd = "targetcli /backstores/%s/ delete %s" % (
+                self.iscsi_backend,
+                self.backstore_name,
+            )
+            process.run(backstore_cmd)
+            # Save configuration
+            process.system("targetcli / saveconfig")
+        else:
+            LOG.info("No backstore %s found. Nothing to do!" % self.backstore_name)
+
+
+class MultiPathsIscsiLIO(IscsiLIO):
+    """
+    Iscsi class supporting multi-paths for the LIO backend.
+    """
+
+    def __init__(self, params):
+        """
+        Initialize multi-paths target in iscsi.
+        """
+        self.target = params.get("target")
+        self.export_flag = False
+        self.luns_in_target = {}
+        self.portal_ip = params.get("portal_ip", ["127.0.0.1"])
+        self.id = params.get("iscsi_thread_id", data_factory.generate_random_string(4))
+        self.iscsi_target_attrs = params.get("attrs", "")
+        self.backstores = params.get("backstores")
+
+        # CHAP AUTHENTICATION
+        self.chap_flag = False
+        self.chap_user = params.get("chap_user")
+        self.chap_passwd = params.get("chap_passwd")
+        self.enable_authentication = params.get("enable_authentication")
+        if self.chap_user and self.chap_passwd:
+            self.chap_flag = True
```

Comment:
I am wondering if we can set different u/p for different targets or portal ips, but never mind, I don't think we have this requirement now